static/app/components/notificationActions/notificationActionItem.tsx

```diff
@@ -0,0 +1,344 @@
+  project: Project;
+  /**
+   * Whether to initially display edit mode
+   * Set to "true" when adding a new notification action
+   */
+  defaultEdit?: boolean;
+  /**
+   * Optional list of roles to display as recipients of Sentry notifications
+   */
+  recipientRoles?: string[];
+};
+
+const NotificationActionItem = ({
+  action,
+  index,
+  availableActions,
+  defaultEdit = false,
+  pagerdutyIntegrations,
+  project,
+  recipientRoles,
+  onDelete,
+  onUpdate,
+}: NotificationActionItemProps) => {
+  const [isEditing, setIsEditing] = useState(defaultEdit);
+  const [editedAction, setEditedAction] = useState(action);
+  const serviceType = action.serviceType;
+  const api = useApi();
+  const organization = useOrganization();
+
+  const renderIcon = () => {
+    switch (serviceType) {
+      // Currently email and Sentry notification use the same icon
+      case NotificationActionService.EMAIL:
+      case NotificationActionService.SENTRY_NOTIFICATION:
+        return <IconMail size="sm" />;
+      default:
+        return <PluginIcon pluginId={serviceType} size={16} />;
+    }
+  };
+
+  const renderDescription = () => {
+    switch (serviceType) {
+      case NotificationActionService.SENTRY_NOTIFICATION:
+        return (
+          <Fragment>
+            <div>{t('Send an email notification to the following roles')}</div>
+            {recipientRoles?.map(role => (
+              <NotificationRecipient key={role}>{role}</NotificationRecipient>
+            ))}
+          </Fragment>
+        );
+      case NotificationActionService.SLACK:
+        return (
+          <Fragment>
+            <div>{t('Send a notification to the')}</div>
+            <NotificationRecipient>{action.targetDisplay}</NotificationRecipient>
+            <div>{t('channel')}</div>
+          </Fragment>
+        );
+      case NotificationActionService.PAGERDUTY:
+        return (
+          <Fragment>
+            <div>{t('Send a notification to the')}</div>
+            <NotificationRecipient>{action.targetDisplay}</NotificationRecipient>
+            <div>{t('service')}</div>
+          </Fragment>
+        );
+      default:
+        // TODO(enterprise): descriptions for email, msteams, sentry_app
+        return null;
+    }
+  };
+
+  const handleDelete = async () => {
+    const endpoint = `/organizations/${organization.slug}/notifications/actions/${action.id}/`;
+    try {
+      await api.requestPromise(endpoint, {
+        method: 'DELETE',
+      });
+      addSuccessMessage(t('Successfully deleted notification action'));
+      onDelete(index);
+    } catch (err) {
+      addErrorMessage(t('Unable to delete notification action'));
+    }
+  };
+
+  const handleCancel = () => {
+    if (action.id) {
+      setIsEditing(false);
+      return;
+    }
+    // Delete the unsaved notification action
+    onDelete(index);
+  };
+
+  const handleSave = async () => {
+    const formProps = getFormProps();
```

Comment:
maybe we can just deconstruct the props?

Reply:
Can we name this something other than props?

Maybe `formData`? or `formState` or something?
props is just kinda funky cuz it's now obfuscated by react :/